lib/psm.c

```diff
@@ -469,6 +469,17 @@ static rpmpsm rpmpsmNew(rpmts ts, rpmte te, pkgGoal goal)
 	    psm->scriptArg = npkgs_installed + 1;
 	    psm->countCorrection = 0;
 	    break;
+	case PKG_PREUNTRANS:
+	    if (rpmteDependsOn(psm->te))
+		psm->scriptArg = npkgs_installed;
+	    else
+		psm->scriptArg = npkgs_installed - 1;
+	    psm->countCorrection = -1;
+	    break;
+	case PKG_POSTUNTRANS:
+	    psm->scriptArg = npkgs_installed;
```

Comment:
Shouldn't this be `npkgs_installed - 1` as it's a removal? (Maybe I'm just understanding this wrong)

Reply:
The npkgs_installed is calculated at the time the script runs, so this all depends on where exactly in the transaction stuff happens. When this is calculated for the other scriptlets, the package hasn't yet been removed, but in postuntrans it has. 

The related countCorrection in both these is actually unused because that only matters for triggers. I should make a comment on that, it's quite confusing.

Reply:
Makes sense, thanks.